sdk/core/core/inc/az_span.h

```diff
@@ -135,161 +158,306 @@ AZ_NODISCARD az_span az_span_from_str(char* str);
 
 /**
- * @brief converts @b src span to zero-terminated str. Content is copied to @b buffer and then \0 is
- * addeed at the end. Then out_result will be created out of buffer
+ * @brief az_span_append appends the bytes referred to by the source span into the
+ * destination span starting at the destination span's length.
  *
- * @param s buffer where to write str
- * @param max_size span to str to create
- * @param span span with zero terminated str
- * @return AZ_NODISCARD az_span_to_str
+ * @param[in] destination The az_span where the bytes should be appended to.
+ * @param[in] source Refers to the bytes to be appended to the destination
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the
+ * destination span with its length updated.
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the /p destination is not big enough to
+ *  contain the appended bytes.
  */
-AZ_NODISCARD az_result az_span_to_str(char* s, int32_t max_size, az_span span);
+AZ_NODISCARD az_result az_span_append(az_span destination, az_span source, az_span* out_span);
 
-AZ_NODISCARD az_result az_span_append_double(az_span span, double value, az_span* out);
+/**
+ * @brief az_span_append_i32toa appends an int32 as digit characters to the destination
+ * starting at the destination span's length.
+ *
+ * @param[in] destination The az_span where the bytes should be appended to
+ * @param[in] source The int32 whose number is appended to the destination span as ASCII
+ * digits
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the
+ * destination span with its length updated
+ * @return An #az_result value indicating the result of the operation.
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the /p destination is not big enough to
+ * contain the appended bytes
+ */
+AZ_NODISCARD az_result
+az_span_append_i32toa(az_span destination, int32_t source, az_span* out_span);
 
-AZ_NODISCARD az_result az_span_append_uint64(az_span* self, uint64_t n);
+/**
+ * @brief az_span_append_u32toa appends a uint32 as digit characters to the destination
+ * starting at the destination span's length.
+ *
+ * @param[in] destination The az_span where the bytes should be appended to
+ * @param[in] source The uint32 whose number is appended to the destination span as ASCII
+ * digits
+ * @param[out] out_span A pointer to an az_span that receives the span referring to the
+ * destination span with its length updated
+ * @return An #az_result value indicating the result of the operation:
+ *          #AZ_OK if successful
+ *          #AZ_ERROR_INSUFFICIENT_SPAN_CAPACITY if the destination is not big enough to
+ * contain the appended bytes
+ */
+AZ_NODISCARD az_result
+az_span_append_u32toa(az_span destination, uint32_t source, az_span* out_span);
```

Comment:
They are named differently because they do different things. *_uint8 actaully appends a single byte as is to the span.  *toa functions convert the number to ascii and then append it to the span; the number is NOT appended as-is.